src/components/PlantCard.js

```diff
@@ -1,18 +1,53 @@
-import React from "react";
+import React, { useState } from "react"
+
+function PlantCard({ plant, updatePlant }) {
+  const { name, image, price } = plant
+  const [soldOut, setSoldOut] = useState(plant.isSoldOut)
+  const [newPrice, setNewPrice] = useState(price)
+
+  const handleSoldOut = (e) => {
+    e.preventDefault()
+    updatePlant("edit", plant, "isSoldOut", !soldOut)
+    setSoldOut(!soldOut)
+  }
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    const inputPrice = e.target.elements.price.value
+    setNewPrice(inputPrice)
+    updatePlant(plant, "price", parseInt(inputPrice))
```

Comment:
Hm... I don't think this is doing anything based on your implementation of `updatePlant`.